parcels/application_kernels/advection.py

```diff
@@ -251,21 +251,21 @@ def compute_ds(F0, F1, r, direction, tol):
     s_min = min(abs(ds_x), abs(ds_y), abs(ds_z), abs(ds_t / (dxdy * dz)))
 
     # calculate end position in time s_min
-    def compute_rs(ds, r, B, delta, s_min):
+    def compute_rs(r, B, delta, s_min):
         if abs(B) < tol:
             return -delta * s_min + r
         else:
             return (r + delta / B) * math.exp(-B * s_min) - delta / B
 
-    rs_x = compute_rs(ds_x, xsi, B_x, delta_x, s_min)
-    rs_y = compute_rs(ds_y, eta, B_y, delta_y, s_min)
+    rs_x = compute_rs(xsi, B_x, delta_x, s_min)
+    rs_y = compute_rs(eta, B_y, delta_y, s_min)
 
-    particle_dlon = (1.-rs_x)*(1.-rs_y) * px[0] + rs_x * (1.-rs_y) * px[1] + rs_x * rs_y * px[2] + (1.-rs_x)*rs_y * px[3] - particle.lon  # noqa
-    particle_dlat = (1.-rs_x)*(1.-rs_y) * py[0] + rs_x * (1.-rs_y) * py[1] + rs_x * rs_y * py[2] + (1.-rs_x)*rs_y * py[3] - particle.lat  # noqa
+    particle_dlon += (1.-rs_x)*(1.-rs_y) * px[0] + rs_x * (1.-rs_y) * px[1] + rs_x * rs_y * px[2] + (1.-rs_x)*rs_y * px[3] - particle.lon  # noqa
+    particle_dlat += (1.-rs_x)*(1.-rs_y) * py[0] + rs_x * (1.-rs_y) * py[1] + rs_x * rs_y * py[2] + (1.-rs_x)*rs_y * py[3] - particle.lat  # noqa
```

Comment:
Was the direct assignment instead of `+=` a bug before?

I'm mystified a bit about `particle_dlon` and `particle_dlat` not being used in the rest of the function, but I'll assume that its a bit of parcels C magic I'm not (yet!) familiar with.